### DISCUSSION TITLE
AnimatedTransformation?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1246] [Ideas]
Hi:

Today we discussed about the 3D model animations
and our group members suggest that 
in each key frame
 we should store the component transformation data rather than the whole data of obj file
so I think this mechanism also apply to 2D animation like skeleton animation and svg animation
and the good part is this animation could reuse the existing code of Animation of FXGL
Is it feasible in this way?
and maybe we could develop an editor for this animated transformation

thx

**Comment by AlmasB** [comment]
This sounds interesting, it would be good to play with some concrete data. Do you have a minimal example file that describes the animation?

**Reply by chengenzhao** [reply]
I do have a sprit in my game
that is also the inspiration of new method of Animation class like playFrom, playTo method should be applied in that class
The very basic transformation is rotated around pivotal point
in the zip file, the launcher itself has an animated texture
and the fire flame should be rotated to cooperate with launcher animation
This could be a nice example for animation texture and transformation
[launcher.zip](https://github.com/AlmasB/FXGL/files/10986001/launcher.zip)

**Reply by AlmasB** [reply]
I wonder if one solution to this is to simply bind certain properties (like angle, position, etc.) of one entity or animated texture to another.
